src/support.ts

```diff
@@ -0,0 +1,43 @@
+import { STATUS_CODE } from "../mod.ts";
+
+const applicationTypeIsJson = (headers: Headers): boolean =>
+  !!headers.get("content-type")?.toLowerCase().includes("application/json");
+
+type Method = "GET" | "PUT" | "PATCH" | "DELETE" | "POST";
+type EnumValues<T> = T[keyof T];
+export type KallResponse<R> = Promise<
+  { status: EnumValues<typeof STATUS_CODE>; body: R; response: Response }
+>;
+
+type Fetch = typeof fetch;
+const supportedFetch: Fetch =
+  (typeof window !== "undefined" || typeof Deno !== "undefined")
+    ? fetch // @ts-ignore To make compatiable with Node.js
+    : require("node-fetch");
+
+export const performRequest = async <T, R>(
+  method: Method,
+  url: string,
+  requestBody: T | null,
+  requestInit: RequestInit,
+): KallResponse<R> => {
+  const response = await supportedFetch(url, {
+    method,
+    body: requestBody ? JSON.stringify(requestBody) : null,
+    headers: {
+      "Content-Type": "application/json",
+      ...requestInit.headers,
+    },
+    ...requestInit,
+  });
+
+  const parsedBody = applicationTypeIsJson(response.headers)
+    ? await response.json() as R
+    : await response.text() as R;
+
+  return {
+    status: response.status as EnumValues<typeof STATUS_CODE>,
+    body: parsedBody,
+    response,
+  };
```

Comment:
I agree with turning this into an object! 
I remember quite liking the array-style at the time, but now I can't recall why.. 

This leaves the same usage pattern, which is what matters. 👍